### ISSUE TITLE
Adding users with roles - Role based access to users

### ISSUE THREAD

**Issue by jr1234** [drhenner/ror_ecommerce#155]
I am unable to add users with at least one role assigned to the user. For example if I don't check any roles, it gets created as a regular user. He doesn't have any other roles. Interestingly in the db table there are no roles assigned to this user

When I check one of the roles (administrator), and click create, it returns back to the same Add User page and user is not created. In the DB the new user is not there as well. The logs say Rollback.

This image is the screen

![screen shot 2015-09-09 at 12 24 19 pm](https://cloud.githubusercontent.com/assets/13705191/9755198/b5cafbe4-56ed-11e5-9104-705709c0ab40.png)

This image is the log showing rollback

![image](https://cloud.githubusercontent.com/assets/13705191/9755237/fff18774-56ed-11e5-8259-ce72ce586f66.png)

Basically I would like to give role based access to users using the default roles you have provided. I am for now unable to make it to work. I would appreciate any help in this direction.

Thanks
Raghu

**Comment by drhenner** [comment]
Raghu,

Try to create without the roles assigned and then update the user with the roles assigned... I think that is a work-around but I'll come up with a fix later

**Comment by jr1234** [comment]
Thank you David for you prompt response. It is very help full.

Thanks

**Comment by drhenner** [comment]
BTW:  the code is fixed now
